### PR TITLE
Fix issue #127

### LINKED ISSUE #127: sbgECom library being linked before declared
If you are generating the project on Windows, the CMakeList in the libraries folder tries to use the sgbECom library on [line 10](https://github.com/uorocketry/rocket-code-2020/blob/aa797b100dc746e02cd8536a9654c11ec979f2cf/libraries/CMakeLists.txt#L10) before it is declared in the same file on [line 22](https://github.com/uorocketry/rocket-code-2020/blob/aa797b100dc746e02cd8536a9654c11ec979f2cf/libraries/CMakeLists.txt#L22).

```console
CMake Error at libraries/CMakeLists.txt:10 (target_link_libraries):
  Cannot specify link libraries for target "sbgECom" which is not built by
  this project.
```

I managed to fix this by moving the `target_link_libraries` to right bellow `add_library`.
```cmake
...
add_library(sbgECom STATIC ${SRC} ${COMMON_SRC} ${SM_SRC})

if (MSVC)
    target_link_libraries(sbgECom Ws2_32)
endif()
...
```

### PR DESCRIPTION
**Pull request by seofernando25**
Fix #127

### CODE REVIEW

**Comment by ajayyy** [comment]
It is a private repo

@sebastiengrd can you add them to the org?

File: libraries/CMakeLists.txt [hunk 1]
```diff
@@ -6,9 +6,7 @@ file(GLOB_RECURSE SRC ${PROJECT_SOURCE_DIR}/libraries/sbgECom/src/*.c)
 file(GLOB_RECURSE COMMON_SRC ${PROJECT_SOURCE_DIR}/libraries/sbgECom/common/*.c)
 file(GLOB_RECURSE SM_SRC ${PROJECT_SOURCE_DIR}/libraries/sbgECom/src/*.cpp)
 
-if (MSVC)
-    target_link_libraries(sbgECom Ws2_32)
-endif()
+
```
Comment (ajayyy) [inline libraries/CMakeLists.txt:9]: Can you remove the extra spacing?

**Comment by ajayyy** [comment]
It seems you are part of the org, can you access https://github.com/uorocketry/sbgECom ?

**Comment by SeoFernando25** [comment]
Yes, I can. 

But why would it fail with `Catch2`, a public library, for example?

```
Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'https://github.com/catchorg/Catch2.git' into submodule path '/home/runner/work/rocket-code-2020/rocket-code-2020/libraries/Catch2' failed
Failed to clone 'libraries/Catch2'. Retry scheduled
Cloning into '/home/runner/work/rocket-code-2020/rocket-code-2020/libraries/WiringPi'...
Load key "/home/runner/.ssh/id_rsa": invalid format
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.
```

**Comment by ajayyy** [comment]
Maybe you have to set an SSH key for github actions to use. @jonapap do you know what's up?

**Comment by jonapap** [comment]
Because the SBG repo is private, we must have a ssh key stored as a secret for the ci. For security reasons, GitHub will not pass any secrets to pull requests originating from outside the repo. Therefore, you will need to push your changes to the main repo and reopen a new pr (or we could just force merge this pr).

Sorry for that. I wish there would be a better way to do this.

**Pull request merged**